Create atan2 entry for golang (#3419)
Codecademy/docs: content/go/concepts/math-functions/terms/atan2/atan2.md

---
Title: 'Atan2()'
Description: 'Returns the arctangent value of the x/y value.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Math'
  - 'Trigonometry'
  - 'Functions'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---

> The **`Atan2()`** function returns the arctangent value of the x/y value.

Reviewer: ```suggestion
The **`Atan2()`** function returns the arctangent value of the *y/x* value.
```

## Syntax

```pseudo
result := math.Atan2(y,x)
```

Notes on `Atan2()`:

> - The ratio of the two arguments `(y/x)` is used to calculate the value of the arc tangent.

Reviewer: ```suggestion
- The ratio of the two arguments `(y/x)` is used to calculate the value of the arctangent.
```

- The `result` will be between between -π/2 to π/2 radians.
- The value returned will be of type `float64`.

> ## Special Cases

Reviewer: ```suggestion
### Special Cases
```

> `NAN:`
> 
> - Atan2(y, NaN) = NaN
> - Atan2(NaN, x) = NaN

Reviewer: ```suggestion
`NAN`:
- *Atan2(y, NaN) = NaN*
- *Atan2(NaN, x) = NaN*
```

Reviewer: Update the others as above

`(+-)0:`

- Atan2(+0, x>=0) = +0
- Atan2(-0, x>=0) = -0
- Atan2(y, +Inf) = +0

`(+-)π:`

- Atan2(+0, x<=-0) = +π
- Atan2(-0, x<=-0) = -π
- Atan2(y>0, -Inf) = +π
- Atan2(y<0, -Inf) = -π

`(+-)π/2:`

- Atan2(y>0, 0) = +π/2
- Atan2(y<0, 0) = -π/2
- Atan2(+Inf, x) = +π/2
- Atan2(-Inf, x) = -π/2

`(+-)π/4:`

- Atan2(+Inf, +Inf) = +π/4
- Atan2(-Inf, +Inf) = -π/4

`(+-)3π/4:`

- Atan2(+Inf, -Inf) = 3π/4
- Atan2(-Inf, -Inf) = -3π/4

## Example

The following code calculates the arctangent and prints out the result:

```go
package main

> import "fmt"
> import  "math"

Reviewer: ```suggestion
import (
  "fmt"
  "math"
  )
```

Reviewer: make sure indentations in the code use 2 spaces not tabs
